Add an optional Device parameter to pin_memory/is_pinned that does nothing

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60547 Make pin_memory/is_pinned use BackendSelect
* **#60201 Add an optional Device parameter to pin_memory/is_pinned that does nothing**
* #59655 Remove some unnecessary functions from CUDAHooks

This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D29331880](https://our.internmc.facebook.com/intern/diff/D29331880)

ezyang committed:
Add an optional Device parameter to pin_memory/is_pinned that does nothing

This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Add an optional Device parameter to pin_memory/is_pinned that does nothing"


This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Add an optional Device parameter to pin_memory/is_pinned that does nothing"


This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29331880).

ezyang commented:
ah, hard coded jit schema, my favorite

ezyang committed:
Update on "Add an optional Device parameter to pin_memory/is_pinned that does nothing"


This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D29331880](https://our.internmc.facebook.com/intern/diff/D29331880)

[ghstack-poisoned]

ezyang committed:
Update on "Add an optional Device parameter to pin_memory/is_pinned that does nothing"


This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D29331880](https://our.internmc.facebook.com/intern/diff/D29331880)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29331880).

bdhirsh commented on aten/src/ATen/native/Memory.cpp:
```diff
@@ -9,14 +9,16 @@
     AT_ERROR("cannot pin '", self.toString(), "' only dense CPU tensors can be pinned");
   }
+  TORCH_CHECK(!device.has_value() || device->is_cuda(), "non-cuda device doesn't have a concept of pinned memory");
   if (self.is_pinned()) {
```
should we be passing `device` into `self.is_pinned()`?

bdhirsh replied:
oh i see you delete this in the next PR and let BackendSelect deal with dispatch, I'll keep reading

ezyang committed:
Update on "Add an optional Device parameter to pin_memory/is_pinned that does nothing"


This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D29331880](https://our.internmc.facebook.com/intern/diff/D29331880)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29331880).

ezyang committed:
Update on "Add an optional Device parameter to pin_memory/is_pinned that does nothing"


This is to flush out BC/FC problems with adding this parameter.  Later
PR will actually add the desired functionality.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D29331880](https://our.internmc.facebook.com/intern/diff/D29331880)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29331880).
